### ISSUE TITLE
Improve info accuracy in the variables panel

### ISSUE THREAD

**Issue by berarma** [cakephp/debug_kit#291]
I'm using debug_kit a lot to learn from my mistakes and I found some cases where the variables panel doesn't tell me the truth, so I have to resort to debug() to know what's hapening.

Here's the cases I've found:
```php
        $this->set('obj', new \stdClass());
        $this->set('resource', opendir('.'));
        $entity = new \Cake\ORM\Entity();
        $entity->errors('test', ['']);
        $this->set(compact('entity'));
```

I know these aren't the most common situations but these are sometimes the cases we need to debug for.

Since I've already done some work in there I can get my hands on it again.

**Comment by berarma** [comment]
I'm experimenting with some changes in the variables panel in my branch: https://github.com/berarma/debug_kit/tree/issue-291

It's not ready for a PR but while I'm testing it I could benefit from your corrections and ideas.

I've improved displaying empty values and added type information and string delimitation. I've also tried to implement the suggestion in #277 just to see how helpful it could be, though I'm not sure it can always be meaningful.

I've created a new method since I don't think this changes could improve the include panel, but could be useful in the request and session panels.

There's some things I couldn't improve in the ToolbarHelper since the problem is somewhere else. I guess since data is saved in sqlite it's being serialized and thus resources and closures get converted to integer and string. There's something happening with entity errors too, I have to investigate it.
